### PR TITLE
Post list cards layout tweaks

### LINKED ISSUE #4285: Reader / Posts: Update margins and spacing to match an 8px grid
Margins currently are 14px and horizontal and vertical spacing within the cards is a bit inconsistent in places.  Make a pass to update spacing to conform to an 8px grid. 

FYI @shaunandrews and @mnt

### LINKED ISSUE #4332: Post List: Some cards on iOS 9 iPad have wrong height, hiding some features
Some of the cards aren't correctly computing there desired height and are ending up a little shorter than they should be.  The result is the date and in some cases maybe part of the summary is being cropped. 

I can't reproduce this on any iPhone, and previous versions of the app running on iOS 7 do not show the same behavior. 

My guess is its a constraint issue and could potentially be resolved in #4285 but making a note of it here to confirm. 

Example:
screen shot 2015-10-01 at 7 11 51 pm

### PR DESCRIPTION
**Pull request by aerych**
Closes #4332 
Closes #4285 

Tweaks margins and spacing in the post cards, and fixes an issue where row height was incorrectly computed, resulting in some content being cropped. 

Pinging @mnt and @shaunandrews for a pixel sanity check. Shaun this is the PR we chatted about (much) earlier that updates the post cards to match the margins in the reader cards.  To get the right spacing between text field I'm taking the delta of the font-size (which seems to match the cap-height) and line-height for each element and then adding to make a multiple of eight.  It seemed like the right approach but some of the vertical spacing looked off so I've fudged it in a few places.  Maybe there is a more reliable way?

The iPad keeps the larger font sizes, line heights, and left/right margins which it had before.

Here is a gallery of screen shots from a test blog showing the updated margins and spacing. @mnt, @shaunandrews please let me know if you see something we need to nudge a bit. 
https://cloudup.com/cJpNDYg2R2M

@diegoreymendez would you be up for the code review in addition to the design review?

Needs review: @diegoreymendez

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Post/WPStyleGuide+Posts.m [hunk 1]
```diff
@@ -133,7 +133,7 @@ + (NSDictionary *)postCardTitleAttributes
 + (NSDictionary *)postCardSnippetAttributes
 {
     CGFloat fontSize = [UIDevice isPad] ? 16.0 : 14.0;
-    CGFloat lineHeight = [UIDevice isPad] ? 24.0 : 21.0;
+    CGFloat lineHeight = [UIDevice isPad] ? 26.0 : 22.0;
```
Comment (diegoreymendez) [inline WordPress/Classes/ViewRelated/Post/WPStyleGuide+Posts.m]: I'd suggest replacing the magic values with constants.

File: WordPress/Classes/ViewRelated/Post/WPStyleGuide+Posts.m [hunk 2]
```diff
@@ -143,7 +143,7 @@ + (NSDictionary *)postCardSnippetAttributes
 + (NSDictionary *)postCardDateAttributes
 {
     CGFloat fontSize = [UIDevice isPad] ? 14.0 : 12.0;
-    CGFloat lineHeight = [UIDevice isPad] ? 21.0 : 18.0;
+    CGFloat lineHeight = [UIDevice isPad] ? 22.0 : 18.0;
```
Comment (diegoreymendez) [inline WordPress/Classes/ViewRelated/Post/WPStyleGuide+Posts.m]: I'd suggest replacing the magic values with constants.

File: WordPress/Classes/ViewRelated/Post/WPStyleGuide+Posts.m [hunk 3]
```diff
@@ -153,7 +153,7 @@ + (NSDictionary *)postCardDateAttributes
 + (NSDictionary *)postCardStatusAttributes
 {
     CGFloat fontSize = [UIDevice isPad] ? 14.0 : 12.0;
-    CGFloat lineHeight = [UIDevice isPad] ? 21.0 : 18.0;
+    CGFloat lineHeight = [UIDevice isPad] ? 22.0 : 18.0;
```
Comment (diegoreymendez) [inline WordPress/Classes/ViewRelated/Post/WPStyleGuide+Posts.m]: I'd suggest replacing the magic values with constants.

**Comment by diegoreymendez** [comment]
@aerych - Is there any mechanism to reproduce the linked issues?  In my tests the interface seems to behave properly.

**Comment by aerych** [comment]
@diegoreymendez The height issue in #4332 should be noticeable on an iPad, especially for cards that contain a featured image. You should see that the date is no longer visible as its been pushed below the action buttons.

**Comment by shaunandrews** [comment]
I ran through a handful of the screenshots to check the spacing. Everything seems to fall in our 8px grid, which is great.

Just a few minor notes:

**Featured Images**

screen shot 2015-10-08 at 1 59 15 pm

That featured image looks like its "floating" there. What do you think about adding a subtle 1px border (#E9EFF3) to the top and bottom, similar to what we do on the web.

**Excerpts**

screen shot 2015-10-08 at 2 01 07 pm

Maybe out of scope here, but one the web we strip HTML from excerpts, showing just a single paragraph. We also use the fade overlay in place of the ellipsis.

**Comment by aerych** [comment]
Thanks @shaunandrews!

> That featured image looks like its "floating" there. What do you think about adding a subtle 1px border (#E9EFF3) to the top and bottom, similar to what we do on the web.

I'm a fan. Let's make that its own issue so we can keep a narrow scope here.

> Maybe out of scope here, but one the web we strip HTML from excerpts, showing just a single paragraph. We also use the fade overlay in place of the ellipsis.

Yah, out of scope here but since you asked :)

All HTML is also stripped out here, but carriage returns are not.  And I'm not sure if it would make sense to remove carriage returns -- anyone writing verse or short poems would probably find it bothersome.  Thoughts?

The fade overlay is something we still need to test. My gut says its going to negatively impact scrolling performance and is therefore a no-go but I want to see what the Time Profiler says. It would be very nice if it worked out.

**Comment by diegoreymendez** [comment]
@aerych - Removing my comments in the code after our private chat.  Nice work!

:shipit:

**Comment by aerych** [comment]
Thanks @diegoreymendez!

**Pull request merged**